### PR TITLE
improved search

### PR DESCRIPTION
**Pull request by HackerUndKoch**

### CODE REVIEW

File: server/mocks/contacts.js [hunk 1]
```diff
@@ -33,8 +33,12 @@ module.exports = function (app) {
 
   app.get('/api/search', function (req, res) {
     var text = req.query.text;
-    var matches = db.filter(contact => contact.name.indexOf(text) > -1);
-
+    var matches = db.filter(contact => {
+      var l = contact.name.indexOf(text.toLowerCase()) > -1;
+      var u = contact.name.indexOf(text.toUpperCase()) > -1;
+      var c = contact.name.indexOf(text.charAt(0).toUpperCase() + text.slice(1)) > -1;
+      return l || u || c;
```
Comment (cburgdorf) [inline server/mocks/contacts.js]: Is the intention to get it case insensitive? If so, this should do it, right?

`var matches = db.filter(contact => contact.name.toLowerCase().indexOf(text.toLowerCase()) > -1);`

Comment (HackerUndKoch) [inline server/mocks/contacts.js]: yes. that should do the same thing. just easier.

Comment (ThomasBurleson) [inline server/mocks/contacts.js]: I prefer @cburgdorf's approach which is more standard and less obfuscated.

**Comment by PascalPrecht** [comment]
This landed in the solution branches based on rc3

Thanks @HackerUndKoch !

File: server/mocks/contacts.js (hunk 1 above)
Comment (ThomasBurleson) [inline server/mocks/contacts.js]: Why are we using bitwise operators instead of boolean operators?

Comment (PascalPrecht) [inline server/mocks/contacts.js]: This is not the case in RC3 branches anymore. I went with Christophs approach

**Pull request closed without merging**